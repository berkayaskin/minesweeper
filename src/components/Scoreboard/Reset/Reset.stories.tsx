import { Meta, Story } from '@storybook/react/types-6-0'
import { Reset, ResetProps } from './Reset'

export default {
  title: 'Scoreboard/Reset',
  component: Reset,
} as Meta

const Template: Story<ResetProps> = (args) => <Reset {...args} />

export const ResetExample = Template.bind({})
