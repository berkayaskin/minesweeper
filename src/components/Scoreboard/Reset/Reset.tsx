import styled from '@emotion/styled'
import { useState } from 'react'

export interface ResetProps {
  /**
   * Reset action handler
   */
  onReset: () => void
}

export const Reset = ({ onReset }: ResetProps) => {
  const [mouseDown, setMouseDown] = useState(false)

  const handleMouseDown = () => {
    setMouseDown(!mouseDown)
  }

  const handleMouseLeave = () => {
    setMouseDown(false)
  }

  return (
    <Button onClick={onReset} onMouseDown={handleMouseDown} onMouseLeave={handleMouseLeave}>
      {mouseDown ? 'ðŸ˜®' : 'ðŸ™‚'}
    </Button>
  )
}

const Button = styled.button`
  cursor: pointer;
  font-size: 1.5vw;
  font-weight: 700;
  border-width: 0.15vw;
  border-style: solid;
  background-color: #d1d1d1;
  border-color: white #9e9e9e #9e9e9e white;
`
