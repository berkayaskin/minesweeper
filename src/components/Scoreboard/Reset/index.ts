export { Reset } from './Reset'
export type { ResetProps } from './Reset'
