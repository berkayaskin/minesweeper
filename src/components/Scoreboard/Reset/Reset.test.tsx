import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { Reset } from './Reset'

describe('Reset', () => {
  const RenderReset = () => <Reset onReset={() => null} />

   it('Reset renders correctly', () => {
     const { asFragment } = render(<RenderReset />)
     expect(asFragment()).toMatchSnapshot()
   })

  it('should render elements with initial state', () => {
    render(<RenderReset />)
    expect(screen.getByText('ðŸ™‚')).toBeInTheDocument()
  })

  it('onReset handler should be called when clicked', () => {
    const onReset = jest.fn()
    render(<Reset onReset={onReset} />)
    screen.getByText('ðŸ™‚').click()
    expect(onReset).toBeCalled()
  })

  it('should change state when onMouseDown and onMouseLeave are called', () => {
    render(<RenderReset />)

    const smile = screen.getByText('ðŸ™‚')
    fireEvent.mouseDown(smile)
    expect(screen.getByText('ðŸ˜®')).toBeInTheDocument()

    const shocked = screen.getByText('ðŸ˜®')
    fireEvent.mouseLeave(shocked)
    expect(smile).toBeInTheDocument()
  })
})

global.React = React
