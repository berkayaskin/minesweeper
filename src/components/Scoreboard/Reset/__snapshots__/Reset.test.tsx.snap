// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Reset renders correctly 1`] = `
<DocumentFragment>
  <button
    class="css-ko2ctx"
  >
    ðŸ™‚
  </button>
</DocumentFragment>
`;
